### PR TITLE
Azure Monitor docs

### LINKED ISSUE #15241: Azure Monitor Documentation
Port the documentation from the plugin repo to Grafana's documentation.

### PR DESCRIPTION
**Pull request by daniellee**
Still have to add images and do some tidying up.

Fixes #15241

### CODE REVIEW

File: docs/sources/features/datasources/azuremonitor.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+- Do not quote parameters: `MetricNames(hg, Microsoft.Network/publicIPAddresses, grafanaIP)`
+
+{{< docs-imagebox img="/img/docs/v60/azuremonitor-service-variables.png" class="docs-image--no-shadow" caption="Nested Azure Monitor Template Variables" >}}
+
+Checkout the [Templating]({{< relref "reference/templating.md" >}}) documentation for an introduction to the templating feature and the different
+types of template variables.
+
+### Azure Monitor Metrics Whitelist
+
+Not all metrics returned by the Azure Monitor API have values. The Grafana datasource has a whitelist to only return metric names if it is possible they might have values. This whitelist is updated regularly as new services and metrics are added to the Azure cloud. You can find the current whitelist [here](https://github.com/grafana/grafana/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/supported_namespaces.ts).
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken link

File: docs/sources/features/datasources/azuremonitor.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+| *MetricNames(aResourceGroup, aNamespace, aResourceName)* | Returns a list of metric names.                                |
+
+Examples:
+
+- Resource Groups query: `ResourceGroups()`
+- Passing in metric name variable: `Namespaces(cosmo)`
+- Chaining template variables: `ResourceNames($rg, $ns)`
+- Do not quote parameters: `MetricNames(hg, Microsoft.Network/publicIPAddresses, grafanaIP)`
+
+{{< docs-imagebox img="/img/docs/v60/azuremonitor-service-variables.png" class="docs-image--no-shadow" caption="Nested Azure Monitor Template Variables" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+The Azure Monitor service provides metrics for all the Azure services that you have running. It helps you understand how your applications on Azure are performing and to proactively find issues affecting your applications.
+
+Examples of metrics that you can get from the service are:
+
+- Microsoft.Compute/virtualMachines - Percentage CPU
+- Microsoft.Network/networkInterfaces - Bytes sent
+- Microsoft.Storage/storageAccounts - Used Capacity
+
+{{< docs-imagebox img="/img/docs/v60/azuremonitor-service-query-editor.png" class="docs-image--no-shadow" caption="Azure Monitor Query Editor" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+### Azure Monitor Metrics Whitelist
+
+Not all metrics returned by the Azure Monitor API have values. The Grafana datasource has a whitelist to only return metric names if it is possible they might have values. This whitelist is updated regularly as new services and metrics are added to the Azure cloud. You can find the current whitelist [here](https://github.com/grafana/grafana/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/supported_namespaces.ts).
+
+### Azure Monitor Alerting
+
+Grafana alerting is supported for the Azure Monitor service. This is not Azure Alerts support. Read more about how alerting in Grafana works [here]({{< relref "alerting/rules.md" >}}).
+
+{{< docs-imagebox img="/img/docs/v60/azuremonitor-alerting.png" class="docs-image--no-shadow" caption="Azure Monitor Alerting" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+
+## Querying the Application Insights Service
+
+{{< docs-imagebox img="/img/docs/v60/appinsights-service-query-editor.png" class="docs-image--no-shadow" caption="Application Insights Query Editor" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+| *AppInsightsMetricNames()*         | Returns a list of metric names.                            |
+| *AppInsightsGroupBys(aMetricName)* | Returns a list of group bys for the specified metric name. |
+
+Examples:
+
+- Metric Names query: `AppInsightsMetricNames()`
+- Passing in metric name variable: `AppInsightsGroupBys(requests/count)`
+- Chaining template variables: `AppInsightsGroupBys($metricnames)`
+
+{{< docs-imagebox img="/img/docs/v60/appinsights-service-variables.png" class="docs-image--no-shadow" caption="Nested Application Insights Template Variables" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+Table queries are mainly used in the Table panel and row a list of columns and rows. This example query returns rows with the 6 specified columns:
+
+```
+AzureActivity 
+| where $__timeFilter()
+| project TimeGenerated, ResourceGroup, Category, OperationName, ActivityStatus, Caller
+| order by TimeGenerated desc
+```
+
+{{< docs-imagebox img="/img/docs/v60/azureloganalytics-service-query-editor.png" class="docs-image--no-shadow" caption="Azure Log Analytics Query Editor" >}}
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Broken image, but I guess you already know that

File: docs/sources/features/datasources/azuremonitor.md [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+
+     If using the `All` option, then check the `Include All Option` checkbox and in the `Custom all value` field type in the following value: `all`. If $myVar has value `all` then the macro will instead expand to `1 == 1`. For template variables with a lot of options, this will increase the query performance by not building a large where..in clause.
+
+### Azure Log Analytics Builtin Variables
+
+There are also some Grafana variables that can be used in Azure Log Analytics queries:
+
+- `$__from` - Returns the From datetime from the Grafana picker. Example: `datetime(2018-06-05T18:09:58.907Z)`.
+- `$__to` - Returns the From datetime from the Grafana picker. Example: `datetime(2018-06-05T20:09:58.907Z)`.
+- `$__interval` - Grafana calculates the minimum time grain that can be used to group by time in queries. More details on how it works [here](http://docs.grafana.org/reference/templating/#the-interval-variable). It returns a time grain like `5m` or `1h` that can be used in the bin function. E.g. `summarize count() by bin(TimeGenerated, $__interval)`
```
Comment (marefr) [inline docs/sources/features/datasources/azuremonitor.md]: Sort of broken link, wrong anchor? Should probable be relative link as well.

**Review by marefr — CHANGES_REQUESTED** [review]
Sorry for the broken images comments. You'll probably already know about them :) Found some broken links that needs to be fixed. I would also want a link from What's new in v6.0 to Using Azure Monitor.

Besides that looks great :+1:

**Comment by marefr** [comment]
For images and updated sass on docs landing page please include changes from branch: https://github.com/grafana/grafana.org/tree/docs_landing_page_v60

File: docs/sources/features/datasources/azuremonitor.md (hunk 2 above)
Comment (daniellee) [reply docs/sources/features/datasources/azuremonitor.md]: Not for me - did you download the latest from grafana.org and build the docker container?

File: docs/sources/features/datasources/azuremonitor.md (hunk 8 above)
Comment (daniellee) [reply docs/sources/features/datasources/azuremonitor.md]: Was from the docs from the repo - missed making that one relative.

File: docs/sources/features/datasources/azuremonitor.md (hunk 2 above)
Comment (daniellee) [reply docs/sources/features/datasources/azuremonitor.md]: Double checked that they are pushed and they were: https://github.com/grafana/grafana.org/tree/master/static/img/docs/v60

**Review by marefr — APPROVED** [review]

**Pull request merged**